src/components/UI/Icons/BasicIcon.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+  <div :class="iconClass">
+    <QuestionMarkRound v-if="name == 'question-mark-round'"></QuestionMarkRound>
+    <StudentInClass v-if="name == 'student-in-class'"></StudentInClass>
+    <Math v-if="name == 'math'"></Math>
+    <Notepad v-if="name == 'notepad'"></Notepad>
+    <Splash v-if="name == 'splash'"></Splash>
+  </div>
+</template>
+
+<script>
+import QuestionMarkRound from "./QuestionMarkRound.vue";
+import StudentInClass from "./StudentInClass.vue";
+import Math from "./Math.vue";
+import Notepad from "./Notepad.vue";
+import Splash from "./Splash.vue";
+
+export default {
+  name: "BasicIcon",
+  components: {
+    QuestionMarkRound,
```

Comment:
This whole `Vue` file is a bit weird to me, though I'm guessing you wrote it based on some thing you've learnt. 

Is this kind of relation between icons based on some design pattern that I can read about online?

Reply:
the loading of SVG icons is still really messed up / no very simple way. I found this method on Reddit and seems like that is what most people are doing: https://www.reddit.com/r/vuejs/comments/s1mxf7/what_do_you_all_use_for_custom_svg_icons_vite_vue/

We were using a package called `inline-svg` in Plio but that doesn't work with composition API.

Reply:
I see. Okay. Then the name `BasicIcon` also makes sense. Can you link to this thread in a comment? Because this whole structure is still very off to me!